src/offline/cable_checks.F90

```diff
@@ -50,141 +52,561 @@ MODULE cable_checks_module
+          index = FINDLOC(parameter_r1, min_val)
+     END IF
+
+     IF (max_val > parameter_range(2)) THEN
+          index = FINDLOC(parameter_r1, max_val)
+     END IF
+
+     IF (index(1) > 0) THEN
+          CALL range_abort(vname, ktau, met, parameter_r1(index(1), index(2)), &
+                parameter_range, index(1), & 
+          patch(index(1))%latitude, patch(index(1))%longitude)
+     END IF
+
+  END SUBROUTINE check_range_r1_d2
+
+  SUBROUTINE check_range_r2_d3(vname, parameter_r2, parameter_range, ktau, met)
+
+     CHARACTER(LEN=*) :: vname
+     INTEGER, INTENT(IN) :: ktau
+     TYPE(met_type), INTENT(IN) :: met
+
+     REAL(r_2), DIMENSION(:,:,:), INTENT(IN) :: parameter_r2
+     REAL, DIMENSION(2), INTENT(IN) :: parameter_range
+
+     INTEGER, DIMENSION(3) :: index
+     REAL(r_2) :: max_val, min_val
+
+     index = 0
+     max_val = MAXVAL(parameter_r2)
+     min_val = MINVAL(parameter_r2)
+
+     IF (min_val < parameter_range(1)) THEN
+          index = FINDLOC(parameter_r2, min_val)
+     END IF
+
+     IF (max_val > parameter_range(2)) THEN
+          index = FINDLOC(parameter_r2, max_val)
+     END IF
+
+     IF (SIZE(index) > 0) THEN
+          CALL range_abort(vname, ktau, met, REAL(parameter_r2(index(0), index(1), index(2))), &
+                parameter_range, index(1), patch(index(1))%latitude, patch(index(1))%longitude)
+     END IF
+     
+  END SUBROUTINE check_range_r2_d3
+
+  SUBROUTINE check_range_r1_d3(vname, parameter_r1, parameter_range, ktau, met)
+
+     CHARACTER(LEN=*) :: vname
+     INTEGER, INTENT(IN) :: ktau
+     TYPE(met_type), INTENT(IN) :: met
+
+     REAL, DIMENSION(:,:,:), INTENT(IN) :: parameter_r1
+     REAL, DIMENSION(2), INTENT(IN) :: parameter_range
+     INTEGER, DIMENSION(3) :: index
+     REAL :: max_val, min_val
+
+     index = 0
+     max_val = MAXVAL(parameter_r1)
+     min_val = MINVAL(parameter_r1)
+
+     IF (min_val < parameter_range(1)) THEN
+          index = FINDLOC(parameter_r1, min_val)
+     END IF
+
+     IF (max_val > parameter_range(2)) THEN
+          index = FINDLOC(parameter_r1, max_val)
+     END IF
+
+     IF (SIZE(index) > 0) THEN
+          CALL range_abort(vname, ktau, met, parameter_r1(index(1), index(2), index(3)), parameter_range, & 
+               index(1), patch(index(1))%latitude, patch(index(1))%longitude)
+     END IF
+
+  END SUBROUTINE check_range_r1_d3
+
+  ! TODO add for integer
+
+  SUBROUTINE constant_check_range(soil, veg, ktau, met)
+   ! Notes
+   ! Done for uncommented types in cable_def_types_mod 
+   ! Commented ones in cable_def_types need to be checked
+   ! Commented ones here need to be range-specified
+   ! In future, it's better to have a callback for a derived type having both ranges and values
+
+     TYPE (soil_parameter_type), INTENT(IN) :: soil
+     TYPE (veg_parameter_type), INTENT(IN) :: veg
+
+     INTEGER, INTENT(IN) :: ktau
+     TYPE(met_type), INTENT(IN) :: met
+
+     ! Soil
+     !! 1D
```

Comment:
`!!` is part of FORD markup and signals a comment line that gets picked up for documentation. So replacing with `~`.

```suggestion
     !~ 1D
```